Doc/library/typing.rst

```diff
@@ -1565,6 +1739,67 @@ These are not used in annotations. They are building blocks for creating generic
    .. versionadded:: 3.10
 
 
+.. class:: TypeAliasType(name, value, *, type_params=())
+
+   The type of type aliases created through the :keyword:`type` statement.
+
+   Example::
+
+      >>> type Alias = int
+      >>> type(Alias)
+      <class 'typing.TypeAliasType'>
+
+   .. versionadded:: 3.12
+
+   .. attribute:: __name__
+
+      The name of the type alias.
+
+      Example::
+
+         >>> type Alias = int
+         >>> Alias.__name__
+         'Alias'
+
+   .. attribute:: __module__
+
+      The module in which the type alias was defined.
+
+      Example::
+
+         >>> type Alias = int
+         >>> Alias.__module__
+         '__main__'
+
+   .. attribute:: __type_params__
+
+      The type parameters of the type alias, or an empty tuple if the alias is
+      not generic.
+
+      Example::
+
+         >>> type ListOrSet[T] = list[T] | set[T]
+         >>> Alias.__type_params__
+         (T,)
+         >>> type NotGeneric = int
+         >>> NotGeneric.__type_params__
+         ()
+
+   .. attribute:: __value__
+
+      The type alias's value. This is lazily evaluated, so names used in the
+      definition of the alias are not resolved until the ``__value__`` attribute
+      is accessed.
+
+      Example::
+
+         >>> type Mutually = Recursive
+         >>> type Recursive = Mutually
+         >>> Mutually.__value__
+         Recursive
+         >>> Recursive.__value__
+         Mutually
```

Comment:
Okay, this example is slightly mind-blowing, especially because you haven't shown in any of the previous REPL snippets that the repr of a `TypeAliasType` instance is just its name 😆

Reply:
Adding that here